### PR TITLE
LEARNER-5587

### PR DESCRIPTION
**Pull request by bbaker6225**
Add rate limiting to the GradeViewSet and CredentialViewSet credentials endpoints.  Please view the comments of [LEARNER-5587](https://openedx.atlassian.net/browse/LEARNER-5587) for a discussion of the chosen thresholds.

### CODE REVIEW

File: credentials/apps/api/v2/views.py [hunk 1]
```diff
@@ -14,11 +17,38 @@
 log = logging.getLogger(__name__)
 
 
+def cred_throttle_handler(exc, context):
```
Comment (BenjiLee) [inline credentials/apps/api/v2/views.py]: use full name `credentials_throttle_handler`

File: credentials/apps/api/v2/views.py [hunk 2]
```diff
@@ -92,6 +122,8 @@ class GradeViewSet(mixins.CreateModelMixin,
                    viewsets.GenericViewSet):
     permission_classes = (UserCredentialPermissions,)
     serializer_class = UserGradeSerializer
+    throttle_classes = (CredentialRateThrottle,)
+    throttle_scope = 'grade'
```
Comment (bbaker6225) [inline credentials/apps/api/v2/views.py]: grade_view

File: credentials/apps/api/v2/views.py [hunk 3]
```diff
@@ -14,11 +17,38 @@
...
+    response = exception_handler(exc, context)
+
+    if isinstance(exc, Throttled):
+        view = context['view']
+        if isinstance(view, CredentialViewSet):
+            view = 'CredentialViewSet'
+        elif isinstance(view, GradeViewSet):
+            view = 'GradeViewSet'
+
+        log.warning('Credentials API endpoint {} is being throttled.'.format(view))
```
Comment (bbaker6225) [inline credentials/apps/api/v2/views.py]: See if possible to find user here

Comment (bbaker6225) [reply credentials/apps/api/v2/views.py]: It is not

File: credentials/apps/api/v2/tests/test_views.py [hunk 4]
```diff
@@ -469,3 +470,90 @@ def test_update(self, method):
...
+                response = self.client.get(list_path)
+                self.assertEqual(response.status_code, 200)
+
+            # Request after limit should NOT be acceptable
+            response = self.client.get(list_path)
+            self.assertEqual(response.status_code, 429)
+            self.assert_throttling_log_correct(log, 'CredentialViewSet')
+
+    def test_grade_view_throttling(self):
+        """ Verify requests are throttled and a message is logged after limit. """
```
Comment (BenjiLee) [inline credentials/apps/api/v2/tests/test_views.py]: Add docstring note about why this test could potentially be flaky

Comment (BenjiLee) [reply credentials/apps/api/v2/tests/test_views.py]: `Note: There is a potential for this test to be flaky in the case the endpoint we are testing slows down 
over time and/or the rate limit is increased.`

File: credentials/apps/api/v2/tests/test_views.py [hunk 5]
```diff
@@ -469,3 +470,90 @@ def test_update(self, method):
...
+        log_capture.check(
+            (
+                'credentials.apps.api.v2.views',
+                'WARNING',
+                'Credentials API endpoint {} is being throttled.'.format(view_set)
+            )
+        )
+
+    def test_credential_view_throttling(self):
+        """ Verify requests are throttled and a message is logged after limit. """
```
Comment (BenjiLee) [inline credentials/apps/api/v2/tests/test_views.py]: `Note: There is a potential for this test to be flaky in the case the endpoint we are testing slows down 
over time and/or the rate limit is increased.`

**Pull request merged**